### ISSUE TITLE
IDEA: Secondary Action on Buttons (A Secondary Action AFTER completing primary action, *NOT* the same as Short Press or Long Hold Function)

### ISSUE THREAD

**Issue by IAmOrion** [koriwi/freedeck-configurator#52]
A Secondary Action AFTER completing primary action, *NOT* the same as Short Press or Long Hold Function.

Can we add a toggle switch for each button and action (So the toggle can be different on Short Press and Long Press is required) that says something like "Return to page x after action" so that in the freedeck code, we can also add the function that allows returning to a page (or just the start page) upon completing an action.

So in other words, once the Short Press (in my case set as Hot Key) has been triggered it then automatically goes back to the start page or a page of choice.

I will copy this idea over to the freedeck code github too, since technically they're tied together as the function needs to be in the code for the configurator to work with

**Comment by koriwi** [comment]
I thought about this functionality. Only less nice. My original idea was that you could choose if when pressing a button on a page if it should go back to start. Didn't like that you have to choose per page.

But i would implement your idea like this.

- have the normal hotkey/text/special_key UI in the configurator
- add a checkbox (go to page x after press)
- add a drop down selection for the target page (start by default)

This is easy and intuitive in the configurator and easy to implement in the firmware!

What do you think

**Comment by IAmOrion** [comment]
That's pretty much exactly what I was thinking! An additional option (I said toggle switch just in keeping with the UI - like the "Invert" toggle) ... and yes, then simple a page selector the same as if you had chosen page as the action :) 
Sounds good to me

**Comment by koriwi** [comment]
live on develop branch
